File: JY/209. Minimum Size Subarray Sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0

        start = 0
        current_sum = 0
        min_length = 1000000

        for end in range(len(nums)):
            current_sum +=nums[end]

            while current_sum >= target:
                min_length = min(end-start+1, min_length)
                current_sum -= nums[start]
                start += 1

        return min_length if min_length != 1000000 else 0

File: JY/209. Minimum Size Subarray Sum.c
int minSubArrayLen(int target, int* nums, int numsSize) {
    int current_sum = 0;
    int min_length = 1000000;
    int start = 0;

    for (int i = 0 ; i < numsSize ; i++){
        current_sum += nums[i];
        while (current_sum >= target){
            min_length = (min_length < (i - start +1)) ? min_length : (i - start +1);            
            current_sum -= nums[start];
            start++;

        }
            
    }
    int result = (min_length==1000000) ? 0 : min_length ;
    return result;
}